pwa-backend/src/main/java/io/growingabit/app/signup/executors/SignupStageExecutor.java

```diff
@@ -43,17 +45,31 @@ public void exec(final StudentDataSignupStage stage) throws SignupStageExecution
     }
   }
 
-  public void exec(StudentEmailSignupStage stage) throws SignupStageExecutionException {
+  public void exec(final StudentEmailSignupStage stage) throws SignupStageExecutionException {
     Preconditions.checkNotNull(stage);
 
-    final String signupStageIndentifier = Settings.getConfig().getString(StudentEmailSignupStage.class.getCanonicalName());
-    final StudentEmailSignupStage userSignupStage = (StudentEmailSignupStage) this.currentuser.getSignupStages().get(signupStageIndentifier).get();
+    final StudentEmailSignupStage userSignupStage = this.currentuser.getSignupStage(StudentEmailSignupStage.class);
 
     userSignupStage.setData(new StudentConfirmationEmail(stage.getData().getEmail()));
     new StudentEmailSignupStageDao().persist(userSignupStage);
 
-    DeferredTaskSendVerificationEmail deferred = new DeferredTaskSendVerificationEmail(stage.getWebSafeKey());
+    final DeferredTaskSendVerificationEmail deferred = new DeferredTaskSendVerificationEmail(stage.getWebSafeKey());
     QueueFactory.getDefaultQueue().add(TaskOptions.Builder.withPayload(deferred));
   }
 
+  public void exec(final WalletSetupSignupStage stage) throws SignupStageExecutionException {
+    Preconditions.checkNotNull(stage);
```

Comment:
Tutti gli exec usano la stessa Precondition, e in caso sollevano un NullPointerEx. Ma il controller che lo chiama, gestisce solo NotFoundEx e SignupStageExecutionException, di conseguenza quella ce la perdiamo e torniamo un 500 al client. Bug?